web/tests/test_grade_link_feature.py

```
@@ -0,0 +1,672 @@
+        )
+        
+        # Should not raise a validation error
+        try:
+            grade_a_plus.clean()
+        except ValidationError:
+            self.fail("ValidationError was raised for grade 'A+' without a comment")
+            
+    def test_unique_together_constraint(self):
+        """Test that a user can only grade a link once."""
+        # Try to create a second grade from the same user for the same link
+        with self.assertRaises(Exception):  # Could be IntegrityError or ValidationError depending on Django version
```

Ruff (0.8.2)

`assertRaises(Exception)` should be considered evil

(B017)